src/data/fintype/basic.lean

```diff
@@ -1775,37 +1775,63 @@ lemma of_injective [infinite β] (f : β → α) (hf : injective f) : infinite 
 lemma of_surjective [infinite β] (f : α → β) (hf : surjective f) : infinite α :=
 ⟨λ I, by { classical, exactI (fintype.of_surjective f hf).false }⟩
 
+end infinite
+
 instance : infinite ℕ :=
 ⟨λ ⟨s, hs⟩, finset.not_mem_range_self $ s.subset_range_sup_succ (hs _)⟩
 
 instance : infinite ℤ :=
 infinite.of_injective int.of_nat (λ _ _, int.of_nat.inj)
 
-instance [infinite α] : infinite (set α) :=
-of_injective singleton (λ a b, set.singleton_eq_singleton_iff.1)
+instance infinite.set [infinite α] : infinite (set α) :=
+infinite.of_injective singleton (λ a b, set.singleton_eq_singleton_iff.1)
 
-instance [infinite α] : infinite (finset α) := of_injective singleton finset.singleton_injective
+instance [infinite α] : infinite (finset α) :=
+infinite.of_injective singleton finset.singleton_injective
 
 instance [nonempty α] : infinite (multiset α) :=
 begin
   inhabit α,
-  exact of_injective (multiset.repeat default) (multiset.repeat_injective _),
+  exact infinite.of_injective (multiset.repeat default) (multiset.repeat_injective _),
 end
 
 instance [nonempty α] : infinite (list α) :=
-of_surjective (coe : list α → multiset α) (surjective_quot_mk _)
+infinite.of_surjective (coe : list α → multiset α) (surjective_quot_mk _)
+
+instance [infinite α] : infinite (option α) :=
+infinite.of_injective some (option.some_injective α)
+
+instance sum.infinite_of_left [infinite α] : infinite (α ⊕ β) :=
```

Comment:
Can't this and `sum.infinite_of_right` cause diamonds? They should probably be `def` instead.

Reply:
No, they're in `Prop`.